src/gpl/src/nesterovPlace.cpp

```diff
@@ -374,6 +385,22 @@ NesterovPlace::updateGradients(
   debugPrint(log_, GPL, "replace", 3, "updateGrad:  WireLengthGradSum: {:g}", wireLengthGradSum_);
   debugPrint(log_, GPL, "replace", 3, "updateGrad:  DensityGradSum: {:g}", densityGradSum_);
   debugPrint(log_, GPL, "replace", 3, "updateGrad:  GradSum: {:g}", gradSum);
+
+  // sometimes wirelength gradient is zero when design is too small
+  if( wireLengthGradSum_ == 0 ) {
+    wireLengthCoefX_ *= 0.5;
+    wireLengthCoefY_ *= 0.5;
+    baseWireLengthCoef_ *= 0.5;
+    debugPrint(log_, GPL, "replace", 3, 
+        "updateGrad:  sum(WL gradient) = 0 detected, trying again with wlCoef: {:g} {:g}", 
+        wireLengthCoefX_, wireLengthCoefY_);
+
+    // update WL forces
+    nb_->updateWireLengthForceWA(wireLengthCoefX_, wireLengthCoefY_);
+
+    // recursive call again with smaller wirelength coef
+    updateGradients(sumGrads, wireLengthGrads, densityGrads);
```

Comment:
What prevents endless recursion?

Reply:
Good point. I'll make two more variables to stop conversions after 10 trials.